wfe2/wfe.go

```diff
@@ -1085,31 +1089,57 @@ func (wfe *WebFrontEndImpl) RevokeCertificate(
 	response.WriteHeader(http.StatusOK)
 }
 
-// Challenge handles POST requests to challenge URLs.
+// ChallengeHandler handles POST requests to challenge URLs of the form /acme/chall-v3/<authorizationID>/<challengeID>.
 // Such requests are clients' responses to the server's challenges.
-func (wfe *WebFrontEndImpl) Challenge(
+func (wfe *WebFrontEndImpl) ChallengeHandler(
 	ctx context.Context,
 	logEvent *web.RequestEvent,
 	response http.ResponseWriter,
 	request *http.Request) {
-	notFound := func() {
+	slug := strings.Split(request.URL.Path, "/")
+	if len(slug) != 2 {
 		wfe.sendError(response, logEvent, probs.NotFound("No such challenge"), nil)
+		return
 	}
+
+	wfe.Challenge(ctx, logEvent, challengePath, response, request, slug[0], slug[1])
+}
+
+// ChallengeHandlerWithAccount handles POST requests to challenge URLs of the form /acme/chall/{regID}/{authzID}/{challID}.
+//
+// [pattern]: https://pkg.go.dev/net/http#hdr-Patterns
+func (wfe *WebFrontEndImpl) ChallengeHandlerWithAccount(
+	ctx context.Context,
+	logEvent *web.RequestEvent,
+	response http.ResponseWriter,
+	request *http.Request) {
 	slug := strings.Split(request.URL.Path, "/")
-	if len(slug) != 2 {
-		notFound()
+	if len(slug) != 3 {
+		wfe.sendError(response, logEvent, probs.NotFound("No such challenge"), nil)
 		return
 	}
-	authorizationID, err := strconv.ParseInt(slug[0], 10, 64)
+	// Note: the regID is currently ignored.
```

Comment:
Consider making this "note" a TODO referencing the tracking bug or a specific child bug